policy/pa.go

```diff
@@ -293,6 +291,37 @@ func ValidDomain(domain string) error {
 	return nil
 }
 
+// ValidWildcardDomain checks that the domain begins with *. followed by a
+// domain that passes ValidDomain. It does NOT ensure that the domain is absent
+// from any PA blocked lists.
+func ValidWildcardDomain(domain string) error {
+	// Names containing more than one wildcard are invalid.
+	if strings.Count(domain, "*") > 1 {
+		return errTooManyWildcards
+	}
+
+	// If the domain has a wildcard character, but it isn't the first most
+	// label of the domain name then the wildcard domain is malformed
+	if !strings.HasPrefix(domain, "*.") {
+		return errMalformedWildcard
+	}
+
+	// The base domain is the wildcard request with the `*.` prefix removed
+	baseDomain := strings.TrimPrefix(domain, "*.")
+
+	// Names must end in an ICANN TLD, but they must not be equal to an ICANN TLD.
+	icannTLD, err := iana.ExtractSuffix(baseDomain)
+	if err != nil {
+		return errNonPublic
+	}
+	// Names must have a non-wildcard label immediately adjacent to the ICANN
+	// TLD. No `*.com`!
+	if baseDomain == icannTLD {
+		return errICANNTLDWildcard
+	}
```

Comment:
Isn't this virtually identical to the checks performed at the bottom of ValidDomain? Since we're running these checks on `baseDomain`, we can just let ValidDomain handle these. The only difference is `errICANNTLD` vs `errICANNTLDWildcard`, which honestly isn't an important distinction as far as I'm concerned.

Reply:
I considered this, but mutating the range variable `domain` to be the value of `baseDomain` in the wildcard clause makes me uncomfortable. Was this what you envisioned?

Reply:
I was just imagining deleting these two stanzas, and letting the call to `return ValidDomain(baseDomain)` on the next line handle it.

Reply:
Just gave this a shot and I'm not a fan of the change to the error message. `Domain name is a wildcard for an ICANN TLD` would become `Domain name needs at least one dot`, which could be a little confusing for someone.